docs/features/async-streams.md

```diff
@@ -56,7 +56,9 @@ async IAsyncEnumerable<int> GetValuesFromServer()
 }
 ```
 
-**open issue**: Design async LINQ
+Just like in iterator methods, there are several restrictions on where a yield statement can appear in async-iterator methods:
+- It is a compile-time error for a `yield` statement (of either form) to appear in the `finally` clause of a `try` statement.
```

Comment:
>`yield` statement (of either form) [](start = 35, length = 34)

Minor point: perhaps just "... a `yield return` or `yield break` ..."

Reply:
Those two lines were copied from the existing C# spec. I'll keep as-is.

---
In reply to: [240409214](https://github.com/dotnet/roslyn/pull/31527#discussion_r240409214) [](ancestors = 240409214)